### PR TITLE
feat: Read EDCSA private keys

### PR DESCRIPTION
**Pull request by johanandren**
Read elliptic curve private keys and more

References #32339

### CODE REVIEW

**Comment by johanandren** [comment]
The test coverage only verifies the bytes can be loaded as keys, but I need to check the loaded private keys actually work as well.

**Comment by jroper** [comment]
This looks great. Yes, I think browsers are lagging when it comes to Ed25519. Ed25519 is current recommended best practice to use for OpenSSH, services that utilise SSH like GitHub and GitLab use it. It is becoming the recommended algorithm for use with JWTs as well. We can switch to using this for the kalix runtime to parse its keys.

**Review by jroper — APPROVED** [review]

File: akka-pki/src/main/scala/akka/pki/pem/DERPrivateKeyLoader.scala [hunk 1]
```diff
@@ -122,9 +129,132 @@ object DERPrivateKeyLoader {
...
+          case unexpected =>
+            throw new IllegalArgumentException(
+              s"Unexpected type of the algorithm, expected object idenfier, was: [${unexpected.getClass}]")
+        }
+      case unexpected =>
+        throw new IllegalArgumentException(
+          s"Unexpected type of the privateKeyAlgorithm, expected object idenfier, was: [${unexpected.getClass}]")
+    }
+
+    // FIXME we can't actually be sure it is RSA, could just as well be ECDSA or Ed25519
```
Comment (jroper) [inline akka-pki/src/main/scala/akka/pki/pem/DERPrivateKeyLoader.scala:197]: I guess this comment is not needed?

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**